static/scripts/loggedin.js

```diff
@@ -23,92 +23,114 @@ function showHideGlobalAnnouncement() {
-//         }
-// 	}
-// }
-
-// function adjustContentWidth(sidebar) {
-//     const contentWrapper = document.querySelector('.content-wrapper');
-//     if (contentWrapper) {
-//         if (sidebar.classList.contains('hidden')) {
-//             contentWrapper.style.paddingLeft = '0px';
-//         } else {
-//             contentWrapper.style.paddingLeft = '255px';
-//         }
-//     }
-// }
-
-// function toggleSidebar() {
-//     const sidebar = document.querySelector('.sidebar');
-//     const overlay = document.querySelector('.overlay');
-
-// 	if (sidebar) {
-// 		if (sidebar.classList.contains('hidden')) {
-//             sidebar.classList.remove('hidden');
-//             sidebar.classList.add('visible');
-
-//             if (window.innerWidth <= 1279) {
-//                 overlay.style.display = "block";
-//             }
-//         } else {
-//             sidebar.classList.remove('visible');
-//             sidebar.classList.add('hidden');
-
-//             if (window.innerWidth <= 1279) {
-//                 overlay.style.display = "none";
-//             }
-//         } 
-
-//         if (window.innerWidth <= 1279) return;
-//         adjustContentWidth(sidebar);
-// 	}
-// }
-
-// function toggleSidebarOnWindowWidth(sidebar) {
-//     const overlay = document.querySelector('.overlay');
-
-//     if (window.innerWidth <= 1279) {
-//         sidebar.classList.remove('visible');
-//         sidebar.classList.add('hidden');
-//     }
-//     if (window.innerWidth >= 1280) {
-//         sidebar.classList.remove('hidden');
-//         sidebar.classList.add('visible');
-//         overlay.style.display = "none";
-//     }
-//     if (sidebar.classList.contains('hidden')) {
-//         overlay.style.display = "none";
-//     }
-//     adjustContentWidth(sidebar);
-// }
-
-// let priorWidth = window.innerWidth;
-// window.addEventListener('resize', () => {
-//     const sidebar = document.querySelector('.sidebar');
-//     const currentWidth = window.innerWidth;
-
-//     if (window.innerWidth <= 1279 && sidebar.classList.contains('visible') && priorWidth <= currentWidth) {
-//         return;
-//     }
-//     if (currentWidth >= 1280 && sidebar.classList.contains('hidden') && priorWidth >= currentWidth) {
-//         return;
-//     }
-//     toggleSidebarOnWindowWidth(sidebar);
-//     priorWidth = currentWidth;
-// });
+function toggleSidebarItemGroup(groupData) {
+    const itemGroup = document.querySelector(`.${groupData.groupName}`);
+	if (itemGroup) {
+		if (itemGroup.classList.contains('show-subgroup')) {
+            if (groupData.childActive) return;
+
+            itemGroup.classList.remove('show-subgroup');
+            itemGroup.classList.add('hide-subgroup');
+        } else {
+            itemGroup.classList.add('show-subgroup');
+            itemGroup.classList.remove('hide-subgroup');
+        }
+	}
+
+    const toggleIcon = document.querySelector('#group-toggle-icon');
```

Comment:
As this is an id, is `#group-toggle-icon` really unique in the document? Or should we rather use a class?